Does not build using gcc 6

Hi,

as you can read in the Debian bug report
   https://bugs.debian.org/811644
idba does not build with gcc version 6 which has stronger type checks:

> g++ -DHAVE_CONFIG_H -I. -I.. -I../src -I../gtest_src   -Wdate-time -D_FORTIFY_SOURCE=2 -Wall -O3 -fopenmp -pthread -g -O2 -fstack-protector-strong -Wformat -Werror=format-security -c -o sequence_reader.o `test -f '../src/sequence/sequence_reader.cpp' || echo './'`../src/sequence/sequence_reader.cpp
> ../src/sequence/sequence_reader.cpp: In member function 'virtual bool FastaReader::ReadRecord(Sequence&, std::__cxx11::string&, std::__cxx11::string&)':
> ../src/sequence/sequence_reader.cpp:50:40: error: cannot convert 'std::istream {aka std::basic_istream<char>}' to 'bool' in return
>      return ReadFasta(*is_, seq, comment);
>                                         ^
> 
> ../src/sequence/sequence_reader.cpp: In member function 'virtual bool FastqReader::ReadRecord(Sequence&, std::__cxx11::string&, std::__cxx11::string&)':
> ../src/sequence/sequence_reader.cpp:55:49: error: cannot convert 'std::istream {aka std::basic_istream<char>}' to 'bool' in return
>      return ReadFastq(*is_, seq, comment, quality);
>                                                  ^
> 
> Makefile:441: recipe for target 'sequence_reader.o' failed

Please note that only the first issue is reported and there might be similar ones later in the build sequence.

Kind regards

       Andreas.

--- a/src/graph/hash_graph.h
+++ b/src/graph/hash_graph.h
@@ -188,8 +188,13 @@ public:
     void clear() { vertex_table_.clear(); num_edges_ = 0; }
 
 private:
+#if __cplusplus >= 201103L
+    HashGraph(const HashGraph &) = delete;
+    const HashGraph &operator =(const HashGraph &) = delete;
+#else
     HashGraph(const HashGraph &);
     const HashGraph &operator =(const HashGraph &);
+#endif
 
     bool GetNextVertexAdaptor(const HashGraphVertexAdaptor &current, HashGraphVertexAdaptor &next)
     {
@@ -428,7 +433,7 @@ inline std::ostream &operator <<(std::ostream &os, HashGraph &hash_graph)
 
 namespace std
 {
-template <> inline void swap(HashGraph &x, HashGraph &y)
+inline void swap(HashGraph &x, HashGraph &y)
 { x.swap(y); }
 }
 

--- a/test/Makefile.am
+++ b/test/Makefile.am
@@ -16,7 +16,7 @@ sequence_unittest_SOURCES = $(top_srcdir)/src/test/sequence_unittest.cpp
 short_sequence_unittest_SOURCES = $(top_srcdir)/src/test/short_sequence_unittest.cpp
 hash_map_unittest_SOURCES = $(top_srcdir)/src/test/hash_map_unittest.cpp 
 
-noinst_PROGRAMS = \
+check_PROGRAMS = \
 	compact_sequence_unittest \
 	hash_table_unittest \
 	bit_edges_unittest \
@@ -30,3 +30,4 @@ noinst_PROGRAMS = \
 	short_sequence_unittest \
 	hash_map_unittest 
 
+TESTS=$(check_PROGRAMS)

